src/controllers/controller.line.js

```diff
@@ -205,7 +205,7 @@ module.exports = function(Chart) {
 				var sumPos = 0,
 					sumNeg = 0;
 
-				for (var i = this.chart.data.datasets.length - 1; i > datasetIndex; i--) {
+				for (var i = 0; i < datasetIndex; i++) {
```

Comment:
All of the datasets are now being passed into the core.controller updater function in reversed order (this ensures the datasets are drawn on top of each other instead of behind each other).  The change here is just compensating for that reversal by traversing down dataset values, instead of up.